libraries/AP_HAL_Linux/OpticalFlow_Onboard.cpp

```diff
@@ -247,5 +346,13 @@ void OpticalFlow_Onboard::_run_optflow()
         _videoin->put_frame(_last_video_frame);
         _last_video_frame = video_frame;
     }
+
+    if (convert_buffer) {
+        free(convert_buffer);
+    }
+
+    if (crop_buffer) {
+        free(crop_buffer);
+    }
```

Comment:
as far as I can see, if we don't have a way to top the thread this code is actually never executed.